hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -460,8 +471,17 @@ private void syncHive() {
    * this constraint.
    */
   public void setupWriteClient() {
+    setupWriteClient(schemaProvider, false);
+  }
+
+  /**
+   * Note that depending on configs and source-type, schemaProvider could either be eagerly or lazily created.
+   * SchemaProvider creation is a precursor to HoodieWriteClient and AsyncCompactor creation. This method takes care of
+   * this constraint.
+   */
+  private void setupWriteClient(SchemaProvider schemaProvider, boolean forceRecreate) {
     LOG.info("Setting up Hoodie Write Client");
-    if ((null != schemaProvider) && (null == writeClient)) {
+    if (forceRecreate || (null != schemaProvider) && (null == writeClient)) {
```

Comment:
wondering if we can just redo the schema registration without forcing recreating the entire client?

Reply:
that is a good question. @bvaradar , any suggestions?

Reply:
@bvaradar ping again :)

Reply:
I agree. We should just do schema re-registration.

Reply:
Actually, I just realized HoodieWriteConfig schema also needs to be updated in this case. Trying to see if we can have a consistent point of creating WriteClient and SchemaProvider